File: play-2.4.x/K-Modules/kmodules/K-Base/app/K/TokenObject.java
package K;

import java.util.HashSet;
import java.util.Set;
import java.util.UUID;

/**
 * Created by kk on 14-1-7.
 */
public class TokenObject {

    public UUID user_id;

    public Set<Integer> roles;

    public String platform = ClientType.pc_browser;

    public TokenObject() {
        user_id = null;
        roles = new HashSet<>();
    }

    public boolean HasRole(int role) {

        return roles.contains(role);
    }

    public String JsonString() {
        return Helper.ToJsonString(this);
    }

    public static TokenObject FromJsonString(String JsonStr) {
        return Helper.FromJsonString(JsonStr, TokenObject.class);
    }

}

File: play-kotlin-2.5.x/PlayKotlin/kmodules/K-Base/app/K/Common/TokenObject.kt
package K.Common

import java.util.HashSet
import java.util.UUID

/**
 * Created by kk on 14-1-7.
 */
class TokenObject {

    var user_id: UUID? = null

    var roles: Set<Int>

    var platform = ClientType.pc_browser

    init {
        user_id = null
        roles = HashSet<Int>()
    }

    fun HasRole(role: Int): Boolean {

        return roles.contains(role)
    }

    fun JsonString(): String {
        return Helper.ToJsonString(this)
    }

    companion object {

        fun FromJsonString(JsonStr: String): TokenObject? {
            return Helper.FromJsonString(JsonStr, TokenObject::class.java)
        }
    }

}
